src/main/java/frc/robot/subsystems/shooter/ShooterIOSim.java

```diff
@@ -0,0 +1,86 @@
+package frc.robot.subsystems.shooter;
+
+import com.ctre.phoenix6.controls.VelocityVoltage;
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.math.controller.SimpleMotorFeedforward;
+import edu.wpi.first.units.Angle;
+import edu.wpi.first.units.MutableMeasure;
+import edu.wpi.first.units.Units;
+import edu.wpi.first.units.Velocity;
+import edu.wpi.first.wpilibj.Timer;
+import lib.motors.TalonFXSim;
+
+public class ShooterIOSim implements ShooterIO {
+    private final TalonFXSim topMotor;
+    private final TalonFXSim bottomMotor;
+
+    private final VelocityVoltage topControl = new VelocityVoltage(0);
+    private final SimpleMotorFeedforward topFeedForward;
+
+    private final VelocityVoltage bottomControl = new VelocityVoltage(0);
+    private final SimpleMotorFeedforward bottomFeedForward;
+
+    public ShooterIOSim() {
+        topMotor =
+                new TalonFXSim(
+                        1, ShooterConstants.GEAR_RATIO, ShooterConstants.MOMENT_OF_INERTIA, 1);
+        bottomMotor =
+                new TalonFXSim(
+                        1, ShooterConstants.GEAR_RATIO, ShooterConstants.MOMENT_OF_INERTIA, 1);
```

Comment:
The gear of ratio and MoI might be different between the two rollers.

Reply:
4a6531a